src/engine/TransitivePathHashMap.cpp

```diff
@@ -14,10 +14,10 @@
 TransitivePathHashMap::TransitivePathHashMap(
     QueryExecutionContext* qec, std::shared_ptr<QueryExecutionTree> child,
     TransitivePathSide leftSide, TransitivePathSide rightSide, size_t minDist,
-    size_t maxDist)
+    size_t maxDist, Graphs activeGraphs)
```

Comment:
If all these inheriting classes have no exclusive state, but only share the members of the base class, just with different implementations,
then you can get rid of the explicitly + redundantly  reimplemented constructors and just to
`using BaseClass::BaseClass` to pull in the constructors of the base.

Reply:
(If true, this comment probably holds for multiple classes that start with `TransitivePath...`